src/dmd/mars.d

```diff
@@ -1361,7 +1361,37 @@ void addDefaultVersionIdentifiers()
     static if (TARGET.Windows)
     {
         if (global.params.mscoff)
+        {
             VersionCondition.addPredefinedGlobalIdent("CRuntime_Microsoft");
+            if (!global.params.mscrtlib || !global.params.mscrtlib[0])
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_None");
+            }
+            else if (FileName.equals(global.params.mscrtlib, "libcmt"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Static");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Release");
+            }
+            else if (FileName.equals(global.params.mscrtlib, "libcmtd"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Static");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Debug");
+            }
+            else if (FileName.equals(global.params.mscrtlib, "msvcrt"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Dynamic");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Release");
+            }
+            else if (FileName.equals(global.params.mscrtlib, "msvcrtd"))
+            {
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Dynamic");
+                VersionCondition.addPredefinedGlobalIdent("MSRuntime_Debug");
+            }
+            else
+            {
+                assert(0, "unexpected mscrtlib");
```

Comment:
There is also at least `msvcrt100` being used for the mingw based libraries used when no VC is installed. As this is verifying user input it should be an error. (That will probably also make the CI output more readable).

Reply:
Oh wow. `msvcrt100`... that's like a decade old.

I'll add a case for that lib. Is the full set available when using mingw, or just that one?
I'm surprised the mingw libs use the MS CRT... thereby expecting to support the MSVC STL ABI? It's super complicated and crap.

Reply:
> Oh wow. msvcrt100... that's like a decade old.

It's the latest version the mingw .def-files support. Please note that "mingw-based" means that the mingw def files are used to generate import libraries for the most important system DLLs (similar to the libs in the platform SDK) and for a distributable C runtime.
I've chosen the version from VS2010 because later versions can be a hassle to install on Win7 or earlier. `msvcrt100` is a shared release library. It is not the usual mingw runtime.

Reply:
Oh I see. Fair enough.
You don't bundle the full set? That is, the static/dynamic release/debug versions?

Reply:
> You don't bundle the full set? That is, the static/dynamic release/debug versions?

Only the dynamic/release version is redistributable.